cocos/serialization/deserialize.ts

```diff
@@ -481,13 +450,37 @@ interface IFileData extends Array<any> {
     [File.DependUuidIndices]: (StringIndex|string)[];
 }
 
-// type Body = Pick<IFileData, File.Instances | File.InstanceTypes | File.Refs | File.DependObjs | File.DependKeys | File.DependUuidIndices>
-type Shared = Pick<IFileData, File.Version | File.SharedUuids | File.SharedStrings | File.SharedClasses | File.SharedMasks>;
-const PACKED_SECTIONS = File.Instances;
-interface IPackedFileData extends Shared {
-    [PACKED_SECTIONS]: IFileData[];
+type IFileData = MapEnum<{
+    [x in keyof IFileDataMap as `${x}`]: IFileDataMap[x];
+}, 11 /* Currently we should manually specify the enumerators count. */>;
+
+type IIntrudedFileDataMap = Omit<IFileDataMap, File.Version> & {
+    [File.Context]: FileInfo & DeserializeContext;
 }
 
+type IIntrudedFileData = MapEnum<{
+    [x in keyof IIntrudedFileDataMap as `${x}`]: IIntrudedFileDataMap[x];
+}, 11 /* Currently we should manually specify the enumerators count. */>;
+
+type IDeserializeInput = IFileData | IIntrudedFileData;
+
+type ISharedData = TupleSlice<IFileData, 1, 5>;
```

Comment:
You mean something like this?:

![image](https://github.com/cocos/cocos-engine/assets/28807867/d34824cc-1b4a-4317-9582-afa102ec210e)

It can't.

Reply:
![image](https://github.com/cocos/cocos-engine/assets/1503156/ee01367f-4c7d-49d7-bf4e-95b626c008f9)
I mean, it's clearer, even if it's a lot of words

Reply:
`type Shared = ...` is actually:

```ts
{
  // Note: no 0: ... 
  1: ...
  2: ...
  ...
  4: ...
}
```

Since it's not a tuple, I can't express it in `IPackedData` like:

```ts
[
  version: number,
  ...shared: ISharedData, // Not possible because ISharedData is not tuple
  sections: ISectionData[]
]
```